src/core/model.js

```diff
@@ -258,9 +259,11 @@ export default class Model extends Object3D {
   }
 
   setAttributes(attributes = {}) {
-    Object.assign(this.attributes, attributes);
-    this._createBuffersFromAttributeDescriptors(attributes);
-    this.setNeedsRedraw();
+    if (Object.keys(attributes).length > 0) {
```

Comment:
Nit: use `null` as default value and then just do if (attributes).

Reply:
See above comments -- this method is called with the return value of `AttributeManager.getChangedAttributes` which is likely an empty object.